internal/db/sqltest/tests/iam/iam_grant.sql

```diff
@@ -0,0 +1,277 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: BUSL-1.1
+
+begin;
+select plan(27);
+select wtt_load('widgets', 'iam');
+
+-- insert canonical_grant with valid resource
+-- validate the resource is set to 'scope'
+prepare insert_grant_scope as
+  insert into iam_grant
+    (canonical_grant)
+  values
+    ('type=scope;others=stuff;');
+select lives_ok('insert_grant_scope');
+select is(
+  (select resource
+     from iam_grant
+    where canonical_grant = 'type=scope;others=stuff;'),
+  'scope',
+  'resource should be set to "scope" by set_resource() trigger'
+);
+
+-- insert canonical_grant with no type
```

Comment:
I think this comment and test case need updating. There can be valid grants that have no type and should be inserted. However, this is an invalid grant in that it 1. has a trailing `;`, and 2. is not a key=value.